src/lib/mina_base/account.ml

```diff
@@ -871,38 +886,43 @@ let gen_with_constrained_balance ~low ~high : t Quickcheck.Generator.t =
 
 let gen_any_vesting_range =
   let open Quickcheck.Generator.Let_syntax in
-  let open Global_slot in
-  (* vesting period must be at least oe to avoid division by zero *)
-  let%bind vesting_period = Int.gen_incl 1 1000 >>| Global_slot.of_int in
+  let open Global_slot_since_genesis in
+  (* vesting period must be at least one to avoid division by zero *)
+  let%bind vesting_period = Int.gen_incl 1 1000 >>| Global_slot_span.of_int in
   let%bind vesting_end = gen_incl (of_int 1) max_value in
   let%map cliff_time = gen_incl (of_int 1) vesting_end in
   (cliff_time, vesting_end, vesting_period)
 
 let gen_with_vesting_period vesting_period =
   let open Quickcheck.Generator.Let_syntax in
-  let open Global_slot in
-  let min_vesting_end = succ vesting_period in
+  let open Global_slot_since_genesis in
+  let min_vesting_end =
+    Global_slot_span.to_uint32 vesting_period
+    |> Unsigned.UInt32.succ |> Global_slot_since_genesis.of_uint32
+  in
```

Comment:
I don't know how many times this pattern occur but is it worth having `succ` in `Global_slot_span` too?

Reply:
It's there already -- I was just mechanically transforming code, and missed the obvious here.